### PR TITLE
feat: set up linter

### PR DESCRIPTION
**Pull request by chrisrabe**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,14 +7,20 @@
...
     "vue-router": "latest"
   },
   "devDependencies": {
-    "sass": "^1.79.4"
+    "@eslint/js": "^9.12.0",
+    "eslint": "^9.12.0",
+    "eslint-plugin-vue": "^9.28.0",
+    "globals": "^15.10.0",
+    "sass": "^1.79.4",
+    "typescript-eslint": "^8.8.1"
```
Comment (SerkZex) [inline package.json:24]: Is it not better to use tailwindcss?

File: components/Footer.vue [hunk 2]
```diff
@@ -6,41 +6,114 @@
...
       <div class="footer-columns">
         <div class="footer-column">
-          <SectionTitle small>Redot Engine</SectionTitle>
-
-          <NuxtLink href="#" class="footer-link">Download</NuxtLink>
-          <NuxtLink href="#" class="footer-link">Documentation</NuxtLink>
-          <NuxtLink href="#" class="footer-link">Source Code</NuxtLink>
+          <SectionTitle small>
+            Redot Engine
+          </SectionTitle>
```
Comment (SerkZex) [inline components/Footer.vue:41]: personally i like when its compact if the everything could be inlined.

File: package.json (hunk 1 above)
Comment (chrisrabe) [reply package.json:24]: For me it is, but this is sass is the established styling tech for this repo. Can't change it unless the repo owner consents to it

File: components/Footer.vue (hunk 2 above)
Comment (chrisrabe) [reply components/Footer.vue:41]: Did a little digging around. Looks like it's caused by this:

https://eslint.vuejs.org/rules/singleline-html-element-content-newline

I can choose to turn it off but it wouldn't auto format components that do need linebreaks

Comment (SerkZex) [reply components/Footer.vue:41]: yes, i think that is fine, we dont need linebreak when its not required.

File: package.json (hunk 1 above)
Comment (SerkZex) [reply package.json:24]: hmm. i know the guy who created the design with figma he likes sass. One could discuss this in another issue, resolving this

**Pull request merged**